youtube_dl/extractor/cda.py

```diff
@@ -75,21 +109,21 @@ def extract_format(page, version):
                 })
             info_dict['formats'].append(f)
             if not info_dict['duration']:
-                info_dict['duration'] = parse_duration(self._search_regex(
-                    r"duration\s*:\s*(\\?[\"'])(?P<duration>.+?)\1",
-                    unpacked, 'duration', fatal=False, group='duration'))
+                info_dict['duration'] = parse_duration(video.get('duration'))
 
         extract_format(webpage, 'default')
 
         for href, resolution in re.findall(
                 r'<a[^>]+data-quality="[^"]+"[^>]+href="([^"]+)"[^>]+class="quality-btn"[^>]*>([0-9]+p)',
                 webpage):
-            webpage = self._download_webpage(
-                href, video_id, 'Downloading %s version information' % resolution, fatal=False)
+            webpage = self._download_webpage(self._BASE_URL + href, video_id,
+                                             'Downloading %s version information'
+                                             % resolution, fatal=False)
```

Comment:
What's the point changing the original indent? With this style you just waste lots of space.